ui/src/createTelemetryReporter.ts

```diff
@@ -37,3 +38,31 @@ export function createTelemetryReporter(ctx: vscode.ExtensionContext): IInternal
 
     return newReporter;
 }
+
+class InternalTelemetryReporter extends TelemetryReporter implements IInternalTelemetryReporter {
```

Comment:
This will not actually add the shared properties to all events (which is necessary), only to those sent via `postEvent`. Should probably internally register a `TelemetryHandler`.